src/trousse/feature_operations.py

```diff
@@ -25,6 +27,135 @@ def is_similar(self, other: "FeatureOperation"):
         raise NotImplementedError
 
 
+class _OperationsList:
+    def __init__(self):
+        self._operations_list = []
+        self._operations_by_column = collections.defaultdict(list)
+
+    def derived_columns_from_col(self, column: str) -> List[str]:
+        """Return name of the columns created by FeatureOperations applied on ``column``
+
+        Parameters
+        ----------
+        column : str
+            The column on which the FeatureOperation has been applied on.
+
+        Returns
+        -------
+        List[str]
+            Name of the columns created by FeatureOperations applied on ``column``.
+        """
+        derived_columns = []
+        operations = self[column]
+
+        for operation in operations:
+            if column in operation.columns and operation.derived_columns is not None:
+                derived_columns.extend(operation.derived_columns)
+
+        return derived_columns
+
+    def original_columns_from_derived_column(self, derived_column: str) -> List[str]:
+        """Return the name of the columns from which ``derived_column`` is generated from.
+
+        Parameters
+        ----------
+        derived_column : str
+            The column that has been generated
+
+        Returns
+        -------
+        List[str]
+            Name of the columns from which ``derived_column`` has been generated from.
+        """
+        operations = self._operations_from_derived_column(derived_column)
+
+        if len(operations) > 1:
+            raise RuntimeError(
+                "Multiple FeatureOperation found that generated column "
+                f"{derived_column}... the pipeline is compromised"
+            )
+        if len(operations) == 0:
+            raise RuntimeError(
+                "No FeatureOperation found that generated column "
+                f"{derived_column}... the pipeline is compromised"
+            )
+
+        return operations[0].columns
+
+    def _operations_from_derived_column(
+        self, derived_column: str
+    ) -> List[FeatureOperation]:
+        """Return the FeatureOperations that generated ``derived_column``
+
+        Parameters
+        ----------
+        derived_column : str
+            The column that has been generated
+
+        Returns
+        -------
+        List[FeatureOperation]
+            FeatureOperations that generated ``derived_column``
+        """
+        return list(
+            filter(
+                lambda op: op.derived_columns is not None
```

Comment:
If I understand correctly, according to the change made at line 112, will you ever end up with `op.derived_columns = None` ?

Reply:
Yes I can, because at line 112 I'm not assigning back [] to the derived columns attribute, it's just to ease the calculations later. I'm not modifying the instance